### PR TITLE
test(hx-file-icon): surf-2008

### PR DESCRIPTION
**Pull request by BY00565233**
## Description

Unit test `<hx-file-icon>`

image

### What are the relevant story cards/tickets? Any additional PRs or other references?

Jira: SURF-2008(unit test case)

## Before you request a review for this PR:

- [x] Did you add component tests for any new code?
- [x] Did you run the component unit tests via `yarn test` to ensure all tests passed?
- [x] Did you include a screenshot of the component tests?
- [x] If you changed/added functionality, did you update the demo page and documentation?
- [x] If needed, did you add or modify the demo test page to test the changed/added functionality?
- [x] Did you assign reviewers?
- [x] In Jira, have you linked to this PR on the ticket(s)?

### CODE REVIEW

File: src/elements/hx-file-icon/index.spec.js [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+        });
+
+    });
+
+    describe('test Shadow DOM', () => {
+        describe('verify render', () => {
+            it('should have a static Shadow DOM', async function () {
+                const component = /** @type { HXFileIconElement } */ await fixture(template);
+
+                // FIXME: temp fix issue submitted https://github.com/open-wc/open-wc/issues/1540
```
Comment (100stacks) [inline src/elements/hx-file-icon/index.spec.js:40]: Please remove this comment.  It isn't needed.

File: src/elements/hx-file-icon/index.spec.js [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+                const shadow = component.shadowRoot;
+                const query = shadow.querySelector(elSelector);
+                const queryId = query.id;
+
+                expect(queryId).to.equal(id);
+            });
+
+            it('markup should contain a #hxBase with an file type <hx-icon>', async () => {
+                const elSelector = 'div#hxFileIcon > hx-icon';
+                const Type = 'file';
```
Comment (100stacks) [inline src/elements/hx-file-icon/index.spec.js:88]: Please make `type` lowercase.

File: src/elements/hx-file-icon/index.spec.js [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+            });
+
+            it('should have a single <slot>', async () => {
+                const component = /** @type { HXFileIconElement } */ await fixture(template);
+                const shadow = component.shadowRoot;
+                const query = shadow.querySelectorAll('slot');
+                const len = query.length;
+
+                expect(len).to.be.equal(0);
+            });
```
Comment (100stacks) [inline src/elements/hx-file-icon/index.spec.js:60]: Please check this test.  Is there a single `slot` or no slot?

File: src/elements/hx-file-icon/index.spec.js [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+    });
+
+    describe('test $onConnect method', () => {
+        it('should be type on connect', async () => {
+            const component = /** @type {HXFileIconElement} */ await fixture(template);
+            const attr = component.hasAttribute('type');
+
+            expect(attr).to.be.false;
+        });
+    });
```
Comment (100stacks) [inline src/elements/hx-file-icon/index.spec.js:139]: You can remove this `describe` block.

File: src/elements/hx-file-icon/index.spec.js [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+
+    describe('test <hx-file-icon> getter and setter methods', () => {
```
Comment (100stacks) [inline src/elements/hx-file-icon/index.spec.js:141]: Please add a test for setting the `extension` attribute.

File: src/elements/hx-file-icon/index.spec.js [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+                expect(queryId).to.equal(id);
+            });
+
+            it('markup should contain a #hxBase with an file type <hx-icon>', async () => {
+                const elSelector = 'div#hxFileIcon > hx-icon';
+                const Type = 'file';
+                const component = /** @type { HXFileIconElement } */ await fixture(template);
+                const shadow = component.shadowRoot;
+                const query = shadow.querySelector(elSelector);
+                const queryId = query.type;
```
Comment (100stacks) [inline src/elements/hx-file-icon/index.spec.js:92]: Please change to variable to `queryType`.

File: src/elements/hx-file-icon/index.spec.js (hunk 3 above)
Comment (BY00565233) [reply src/elements/hx-file-icon/index.spec.js:60]: As we are not using the slot in shadow.html file, So I completely removed slot section.

File: src/elements/hx-file-icon/index.spec.js (hunk 5 above)
Comment (BY00565233) [reply src/elements/hx-file-icon/index.spec.js:141]: Added the extension attribute test case , Please review

File: src/elements/hx-file-icon/index.spec.js (hunk 6 above)
Comment (BY00565233) [reply src/elements/hx-file-icon/index.spec.js:92]: Changes have been done, Please review

File: src/elements/hx-file-icon/index.spec.js (hunk 4 above)
Comment (BY00565233) [reply src/elements/hx-file-icon/index.spec.js:139]: Changes have been done, Please review

File: src/elements/hx-file-icon/index.spec.js (hunk 2 above)
Comment (BY00565233) [reply src/elements/hx-file-icon/index.spec.js:88]: Changes have been done, Please review

File: src/elements/hx-file-icon/index.spec.js (hunk 1 above)
Comment (BY00565233) [reply src/elements/hx-file-icon/index.spec.js:40]: Changes have been done, Please review

**Pull request merged**